docs/ha-considerations.md

```diff
@@ -0,0 +1,337 @@
+
+```bash
+# /etc/haproxy/haproxy.cfg
+#---------------------------------------------------------------------
+# Global settings
+#---------------------------------------------------------------------
+global
+    log /dev/log local0
+    log /dev/log local1 notice
+    daemon
+
+#---------------------------------------------------------------------
+# common defaults that all the 'listen' and 'backend' sections will
+# use if not designated in their block
+#---------------------------------------------------------------------
+defaults
+    mode                    http
+    log                     global
+    option                  httplog
+    option                  dontlognull
+    option http-server-close
+    option forwardfor       except 127.0.0.0/8
+    option                  redispatch
+    retries                 1
+    timeout http-request    10s
+    timeout queue           20s
+    timeout connect         5s
+    timeout client          20s
+    timeout server          20s
+    timeout http-keep-alive 10s
+    timeout check           10s
+
+#---------------------------------------------------------------------
+# apiserver frontend which proxys to the masters
+#---------------------------------------------------------------------
+frontend apiserver
+    bind *:${APISERVER_DEST_PORT}
+    mode tcp
+    option tcplog
+    default_backend apiserver
+
+#---------------------------------------------------------------------
+# round robin balancing for apiserver
+#---------------------------------------------------------------------
+backend apiserver
+    option httpchk GET /healthz
+    http-check expect status 200
+    mode tcp
+    option ssl-hello-chk
+    balance     roundrobin
+        server ${HOST1_ID} ${HOST1_ADDRESS}:${APISERVER_SRC_PORT} check
+        # [...]
+```
+Again, there are some placeholders in `bash` variable style to expand:
+- `${APISERVER_DEST_PORT}` the port through which Kubernetes will talk to the API Server.
+- `${APISERVER_SRC_PORT}` the port used by the API Server instances
+- `${HOST1_ID}` a symbolic name for the first load-balanced API Server host
+- `${HOST1_ADDRESS}` a resolvable address (DNS name, IP address) for the first load-balanced API Server host
+- additional `server` lines, one for each load-balanced API Server host
+
+### Option 1: Run the services on the operating system
+
+In order to run the two services on the operating system, the respective distribution's package manager can be used to install the software. This can make sense if they will be running on dedicated hosts not part of the Kubernetes cluster.
+
+On a RedHat-style Linux distribution this will most likely look like this:
+```
+# yum install keepalived haproxy
+```
+Having now installed the abovementioned configuration, the services can be enabled and started. On a recent RedHat-based system, `systemd` will be used for this:
+```
+# systemctl enable haproxy --now
+# systemctl enable keepalived --now
```

Comment:
given kubeadm already requires systemd on Linux, i think it's fine to leave the systemd examples